lib/cookie-store/cookie-store.js

```diff
@@ -0,0 +1,350 @@
+'use strict'
+
+const { createDeferredPromise } = require('../fetch/util')
+const { kEnumerableProperty } = require('../core/util')
+const { queryCookies, setCookie, deleteCookie } = require('./util')
+const { parseSetCookie, parseUnparsedAttributes } = require('./parse')
+const { toUSVString } = require('util')
+
+const kCookieStoreConstructable = Symbol('CookieStore constructable')
+
+class CookieStore extends EventTarget {
+  #cookieStore = []
```

Comment:
avoid private properties, they are slow in Node v16 (should be ok in v18), prefer "private" symbols.

Reply:
Ever since switching to v18, I've been using them everywhere now that they're fast! I'll change it to a symbol